rust/connlib/snownet/src/allocation.rs

```diff
@@ -1470,6 +1471,30 @@ mod tests {
         )
     }
 
+    #[test]
+    fn allocation_refresh_timeout_is_reset_after_refresh_request_is_sent() {
+        let mut allocation = Allocation::for_test(Instant::now());
+
+        let allocate = allocation.next_message().unwrap();
+        allocation.handle_test_input(
+            &allocate_response(&allocate, &[RELAY_ADDR_IP4, RELAY_ADDR_IP6]),
+            Instant::now(),
+        );
+
+        let next_timeout = allocation.poll_timeout().unwrap();
+        assert_eq!(next_timeout, allocation.refresh_allocation_at().unwrap());
+
+        allocation.handle_timeout(allocation.poll_timeout().unwrap());
+
+        assert!(allocation.refresh_allocation_at().is_none());
```

Comment:
This is testing internals, I'd like to avoid that.

Can we instead test that we don't run into the refresh loop but just use backoff to retry the refresh? In fact, we should give up retrying the refresh if we have reach the allocation lifetime.

There is probably another test missing here where we should check that the allocation simply is considered gone if we don't refresh it (from the perspective of this component, we don't necessarily get called with `handle_timeout` in time, i.e. the next time might be after the lifetime).